### PR TITLE
RQ-126: persona recommendation experiment

### PR DESCRIPTION
**Pull request by rohanmathur91**
Closes issue: 

## 📜 Summary of changes:
Screenshot 2023-04-04 at 7 59 37 PM

## ✅ Checklist:

- [x] Make sure linting and unit tests pass.
- [x] No install/build warnings introduced.
- [x] Verified UI in browser.
- [x] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/components/misc/PersonaSurvey/ModalFooter.tsx [hunk 1]
```diff
@@ -75,9 +85,23 @@ export const SurveyModalFooter: React.FC<FooterProps> = ({ page }) => {
           APP_CONSTANTS.GA_EVENTS.ATTR.NUMBER_OF_EMPLOYEES,
           userPersona.numberOfEmployees
         );
-        if (window.location.href.includes(PATHS.SHARED_LISTS.VIEWER.RELATIVE)) {
+        if (isSharedListUser) {
           //don't show recommendation screen for shared list users
           dispatch(actions.updateIsPersonaSurveyCompleted(true));
+          dispatch(actions.updatePersonaSurveyPage(page + 1));
```
[the remark this answers is not in the document]
Comment (rohanmathur91) [reply app/src/components/misc/PersonaSurvey/ModalFooter.tsx:91]: I think it's better to update the survey state, if we have any checks dependent on page than that won't be affected

File: app/src/layouts/DashboardLayout/index.js [hunk 2]
```diff
@@ -39,10 +40,17 @@ const DashboardLayout = () => {
...
+    location?.state?.src === "persona_survey_modal";
+
   return (
     <>
       <Layout className="hp-app-layout">
-        {isPricingPage() || isGoodbyePage() || isInvitePage() ? null : (
+        {isPricingPage() ||
+        isGoodbyePage() ||
+        isInvitePage() ||
+        isPersonaRecommendationFeatureflagOn ? null : (
```
Comment (RuntimeTerror10) [inline app/src/layouts/DashboardLayout/index.js:53]: Can we extract these checks at one place?

```js
const shouldRenderSidebar = !isPricingPage() && !isGoodbyePage() && !isInvitePage() && !isPersonaRecommendationFeatureflagOn;

```

```js
{shouldRenderSidebar && <Sidebar visible={visible} setVisible={setVisible} />}
```

**Review by RuntimeTerror10 — CHANGES_REQUESTED** [review]

File: app/src/components/misc/PersonaSurvey/ModalFooter.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (RuntimeTerror10) [reply app/src/components/misc/PersonaSurvey/ModalFooter.tsx:91]: If the survey is completed then the modal will not mount itself

[the remark this answers is not in the document]
Comment (rohanmathur91) [reply app/src/components/misc/PersonaSurvey/ModalFooter.tsx:79]: okay

**Review by RuntimeTerror10 — CHANGES_REQUESTED** [review]

File: app/src/layouts/DashboardLayout/DashboardContent.js [hunk 3]
```diff
@@ -68,6 +74,29 @@ const DashboardContent = () => {
 
   const prevProps = usePrevious({ location });
 
+  useEffect(() => {
+    if (
+      isPersonaRecommendationFlagOn &&
+      userPersona.page === 1 &&
```
[the remark this answers is not in the document]
Comment (rohanmathur91) [reply app/src/layouts/DashboardLayout/DashboardContent.js:80]: missed it

[the remark this answers is not in the document]
Comment (rohanmathur91) [reply app/src/layouts/DashboardLayout/DashboardContent.js:80]: missed it.

**Review by RuntimeTerror10 — APPROVED** [review]

**Pull request merged**